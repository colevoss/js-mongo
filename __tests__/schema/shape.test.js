const Schema = require('../../src/Schema');

describe('Schema.shape', () => {
  const schema = Schema.shape({
    key1: Schema.string,
    key2: Schema.number,
  });

  test('Returns valid if state matches', () => {
    const testData = {
      test: {
        key1: 'test',
        key2: 1,
      },
    };

    expect(schema('test', testData)).toEqual(testData.test);
  });

  test('Returns error', () => {
    const testData = {
      test: {
        key1: 1,
        key2: 'string',
      },
    };

    const checked = schema('test', testData);

    expect(checked).toBeInstanceOf(Error);
    expect(checked).toMatchSnapshot();
  });

  test('__type', () => {
    expect(Schema.shape({}).__type).toBe('shape');
  });

  test('Throws error if an object is not supplied', () => {
    expect(() => Schema.shape()).toThrowErrorMatchingSnapshot();
  });
});
