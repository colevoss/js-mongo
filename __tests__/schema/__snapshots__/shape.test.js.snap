// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema.shape Returns error 1`] = `[Error: TypeError: Expected key1 to be of type string but recieved number of 1,TypeError: Expected key2 to be of type number but recieved string of string]`;

exports[`Schema.shape Throws error if an object is not supplied 1`] = `"Please provide a plain object to Schema.shape."`;
